react/VirtualComponent.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { ReactElement, FC, useMemo } from 'react'
+
+import ComponentGetter from './components/ExtensionPoint/ComponentGetter'
+import { useTreePath, useRuntime } from './core/main'
+import { flatObj, transformLeaves } from './utils/object'
+
+interface Props {
+  virtualTreeId: string
+  props?: Record<string, unknown>
+}
+
+interface RenderVirtualArgs {
+  runtime: any
+  treePath: string
+  hydration: Hydration
+  tree: VirtualTree
+  key?: string
+}
+
+function getVirtualTree(virtualTreeId: string) {
+  return window.__RUNTIME__.virtualTrees[virtualTreeId]
+}
```

Comment:
How about moving this to `utils/assets`, along with `getImplementation`, for example?
hahahah this is more of a question than a change request. What do you think?

Reply:
Agreed!!